cmd/kubeadm/app/util/config/upgradeconfiguration.go

```diff
@@ -137,6 +137,11 @@ func LoadUpgradeConfigurationFromFile(cfgPath string, _ LoadOrDefaultConfigurati
 		}
 	}
 
+	// If no UpgradeConfiguration kind was found in the config file, return an error.
+	if upgradeCfg == nil {
+		return nil, errors.Errorf("no UpgradeConfiguration kind was found in the YAML file")
```

Comment:
```suggestion
		return nil, errors.Errorf("no % found in the configuration file", constants.UpgradeConfigurationKind)
```

let't use the constant and not mention 'kind'.
the file can be json as well.

we should also chanange this i tihnk;
```
	// Check is there any component configs defined in the config file.
	for gvk := range docmap {
		if componentCfgGV.Has(gvk.Group) {
			klog.Warningf("[config] WARNING: YAML document with Component Configs %v is deprecated for upgrade and will be ignored \n", gvk.Group)
			continue
		}
	}
```

i think the logic should be that any kind other than upgradeconfiguration produce the warning?
other places like 'init' have similar i believe.

Reply:
we need a message for this one for unsupported kinds
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/util/config/initconfiguration.go#L361

Reply:
Updated, PTAL. @neolit123

Reply:
>the file can be json as well.

We mentioned `YAML document` in several places, and maybe it should be renamed uniformly with another PR, but that's not a serious issue.

Reply:
> > the file can be json as well.
> 
> We mentioned `YAML document` in several places, and maybe it should be renamed uniformly with another PR, but that's not a serious issue.

agreed, also given this one has to be backported and the YAML issue can be fixed only on master branch.